### ISSUE TITLE
[mypy] Incompatible types in assignment

### ISSUE THREAD

**Issue by foarsitter** [dapper91/pydantic-xml#65]
Thanks for providing this extension, it makes working with xml fun again. 

The `from_*` methods are typed against `BaseXmlModel` which results in an mypy error. 

```
message: Message | None = Message.from_xml(xml)
```

```
error: Incompatible types in assignment (expression has type "BaseXmlModel | None", variable has type "Message | None")
```

Pydantic itself [uses](https://github.com/pydantic/pydantic/blob/90326480ff1061f690aa193bd112cd17a6d90a70/pydantic/main.py#L90) a `TypeVar` bound to `BaseModel` so children are accepted as return type. 

Is this something you want to support? If so, I can submit a PR.

**Comment by dapper91** [comment]
@foarsitter Hi. Fixed it in version 0.7.0.

**Comment by foarsitter** [comment]
Great work @dapper91, thanks!
